### ISSUE TITLE
pytorch fails to build on darwin

### ISSUE THREAD

**Issue by mstone** [NixOS/nixpkgs#94241]
**Describe the bug**

pytorch is failing to build on darwin due to errors generated by the postFixup script:

https://hydra.nixos.org/build/124421153/nixlog/3

Removing this script via `overridePythonAttrs` makes things work for me locally:

```
$ cat flake.nix
{
  description = "A very basic flake";

  outputs = { self, nixpkgs }: rec {

    overlay = final: prev: rec {
    };

    packages.x86_64-darwin.pt = with import nixpkgs { 
      system = "x86_64-darwin"; 
      overlays = [ overlay ]; 
    }; with pkgs; stdenv.mkDerivation {
      pname = "pt";
      version = "1.0";
      src = null;
      buildInputs = [ 
        (python37.withPackages (ps: with ps; [ 
          bpython
          (pytorch.overridePythonAttrs (old: {postFixup = "";})) 
        ]))
      ];
    };

    defaultPackage.x86_64-darwin = self.packages.x86_64-darwin.pt;

  };
}
```

@stites, @jonringer: git history suggests that you introduced this script in the 1.0.0 -> 1.2.0 transition. Do you know if something like this script is still needed with contemporary versions of pytorch + relevant build packaging?

Also cc/ @teh, @thoughtpolice, @tscholak

**Comment by jonringer** [comment]
I'm not familiar with this, and it looks like @stites  added it for a previous version of pytorch. It will likely need to be modified for it to work with the latest version.

I don't have the hardware to experiment on this issue

**Comment by stites** [comment]
I don't run OSX (the contribution was by @tscholak) but I imagine that these can be removed since, IIRC, the lib names had changed. This `postFixup` is just to get access to library files for downstream projects like hasktorch.

Probably you can keep these lines: https://github.com/NixOS/nixpkgs/blob/c67ba72dd837320da7078df6fe354e4639cdf76c/pkgs/development/python-modules/pytorch/default.nix#L251-L253

**Comment by wegank** [comment]
Fixed by #177812.
